File: 1046-last-stone-weight/1046-last-stone-weight.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        priority_queue<int> mxhp;
        for(int i=0;i<stones.size();i++){
            mxhp.push(stones[i]);            
        }
        while(mxhp.size()>1){
            int f = mxhp.top();
            mxhp.pop();
            int s = mxhp.top();
            mxhp.pop();
            if(f>s){
                mxhp.push(f-s);
            }
        }
        if(mxhp.empty()){
            return 0;
        }
        return mxhp.top();
      
    }
};

File: 1046-last-stone-weight/1046-last-stone-weight.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        stones = [-x for x in stones]
        heapq.heapify(stones)
        while len(stones)>1:
            temp1 = heapq.heappop(stones)
            temp2 = heapq.heappop(stones)
            if temp1 !=temp2:
                heapq.heappush(stones,-abs(temp1-temp2))
        return abs(stones[0]) if len(stones) else 0